pandas/core/arrays/categorical.py

```diff
@@ -50,7 +50,14 @@
 from pandas.core import ops
 from pandas.core.accessor import PandasDelegate, delegate_names
 import pandas.core.algorithms as algorithms
-from pandas.core.algorithms import factorize, take, take_1d, unique1d
+from pandas.core.algorithms import (
+    _get_data_algo,
```

Comment:
side issue, we should probably not use these as private functions, rather should just make them public (or actually hide them and make an interface); could make an issue about this.

Reply:
yah, ive got a branch going that de-privatizing things like this.  holding off since i have too many PRs as it is